c/cert/src/rules/ERR33-C/DetectAndHandleStandardLibraryErrors.ql

```diff
@@ -0,0 +1,275 @@
+/**
+ * @id c/cert/detect-and-handle-standard-library-errors
+ * @name ERR33-C: Detect and handle standard library errors
+ * @description Detect and handle standard library errors. Undetected failures can lead to
+ *              unexpected or undefined behavior.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/err33-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import semmle.code.cpp.commons.NULL
+import codingstandards.cpp.ReadErrorsAndEOF
+
+/**
+ * Classifies error returning function calls based on the
+ * type and value of the required checked
+ */
+abstract class ExpectedErrReturn extends FunctionCall {
+  Expr errValue;
+  ComparisonOperation errOperator;
+
+  Expr getErrValue() { result = errValue }
+
+  ComparisonOperation getErrOperator() { result = errOperator }
+}
+
```

Comment:
this is fairly easy to read as is but maybe add a quick ql dock comment on each with the sample syntax it covers